### PR TITLE
"Copy from course" workflow

### PR DESCRIPTION
**Pull request by the-daniel-rothig**
### Context

As a simple version of templating, allow people to copy across content from other courses when editing courses.

### Changes proposed in this pull request

- List courses with enrichments in the sidebar of the course edit screens
- only nonempty fields applicable to the current edit screen are copied over
- the copy logic lives in the CopyFrom() method on `ICourseEnrichmentViewModel` 

### Guidance to review

n/a

### CODE REVIEW

File: src/ui/Views/Course/About.cshtml [hunk 1]
```diff
@@ -86,6 +86,7 @@
     </div>
     <div class="govuk-grid-column-one-third">
       <aside class="related">
+        @Html.Partial("CopyCourseContent")
```
Comment (defong) [inline src/ui/Views/Course/About.cshtml]: Can you change it to the async version

File: src/ui/Views/Course/About.cshtml [hunk 2]
```diff
@@ -86,6 +86,7 @@
...
         @Html.Partial("FormattingAdvice")
```
Comment (defong) [inline src/ui/Views/Course/About.cshtml]: Can you change it to the async version

File: src/ui/Views/Course/Fees.cshtml [hunk 3]
```diff
@@ -108,6 +108,7 @@
     </div>
     <div class="govuk-grid-column-one-third">
       <aside class="related">
+        @Html.Partial("CopyCourseContent")
```
Comment (defong) [inline src/ui/Views/Course/Fees.cshtml]: Can you change it to the async version

File: src/ui/Views/Course/Fees.cshtml [hunk 4]
```diff
@@ -108,6 +108,7 @@
...
         @Html.Partial("FormattingAdvice")
```
Comment (defong) [inline src/ui/Views/Course/Fees.cshtml]: Can you change it to the async version

File: src/ui/Views/Course/Requirements.cshtml [hunk 5]
```diff
@@ -62,6 +62,7 @@
     </div>
     <div class="govuk-grid-column-one-third">
       <aside class="related">
+        @Html.Partial("CopyCourseContent")
```
Comment (defong) [inline src/ui/Views/Course/Requirements.cshtml]: Can you change it to the async version

File: src/ui/Views/Course/Requirements.cshtml [hunk 6]
```diff
@@ -62,6 +62,7 @@
...
         @Html.Partial("FormattingAdvice")
```
Comment (defong) [inline src/ui/Views/Course/Requirements.cshtml]: Can you change it to the async version

File: src/ui/Views/Course/Salary.cshtml [hunk 7]
```diff
@@ -67,6 +67,7 @@
     </div>
     <div class="govuk-grid-column-one-third">
       <aside class="related">
+        @Html.Partial("CopyCourseContent")
```
Comment (defong) [inline src/ui/Views/Course/Salary.cshtml]: Can you change it to the async version

File: src/ui/Views/Course/Salary.cshtml [hunk 8]
```diff
@@ -67,6 +67,7 @@
...
         @Html.Partial("FormattingAdvice")
```
Comment (defong) [inline src/ui/Views/Course/Salary.cshtml]: Can you change it to the async version

File: src/ui/ViewModels/CourseEnrichmentFieldNames.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        public const string HowSchoolPlacementsWork = "How school placements work";
+        public const string FeeDetails = "Fee details";
+        public const string FeesUkEu = "Fees (UK and EU)";
+        public const string FeesInternational = "Fees (International)";
+        public const string FinancialSupport = "Financial support";
+        public const string Qualifications = "Qualifications needed";
+        public const string PersonalQualities = "Personal qualities";
+        public const string OtherRequirements = "Other requirements";
+        public const string AboutTheTrainingProvider = "About the provider";
+        public const string AboutTheTrainingProvider = "About the provider";
```
Comment (defong) [inline src/ui/ViewModels/CourseEnrichmentFieldNames.cs]: Dupe broke build

File: src/ui/Views/Shared/CopyCourseContent.cshtml [hunk 10]
```diff
@@ -1,5 +1,6 @@
 @{
   IEnumerable<GovUk.Education.ManageCourses.ApiClient.Course> copyables = ViewBag.CopyableCourses as IEnumerable<GovUk.Education.ManageCourses.ApiClient.Course> ?? new List<GovUk.Education.ManageCourses.ApiClient.Course>();
```
[the remark this answers is not in the document]
Comment (the-daniel-rothig) [reply src/ui/Views/Shared/CopyCourseContent.cshtml]: as discussed in meatspace, premature optimisation because `ApiClient.Course` is sufficiently thin

**Pull request merged**